Read CSV file with pandas

In [1]:
from urllib.request import urlretrieve

In [2]:
italy_covid_url = 'https://gist.githubusercontent.com/aakashns/f6a004fa20c84fec53262f9a8bfee775/raw/f309558b1cf5103424cef58e2ecb8704dcd4d74c/italy-covid-daywise.csv'

urlretrieve(italy_covid_url, 'italy-covid-daywise.csv')

('italy-covid-daywise.csv', <http.client.HTTPMessage at 0x2769a669ac0>)

We can read CSV files using a special pandas method

In [11]:
import sys
!{sys.executable} -m pip install pandas

  Using cached pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)
  Using cached tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
   ---------------------------------------- 0.0/11.5 MB ? eta -:--:--
   ---------------------------------------- 0.0/11.5 MB 660.6 kB/s eta 0:00:18
    --------------------------------------- 0.2/11.5 MB 2.1 MB/s eta 0:00:06
   ---- ----------------------------------- 1.3/11.5 MB 10.1 MB/s eta 0:00:02
   -------- ------------------------------- 2.6/11.5 MB 14.9 MB/s eta 0:00:01
   ---------------- ----------------------- 4.6/11.5 MB 21.0 MB/s eta 0:00:01
   ---------------------- ----------------- 6.6/11.5 MB 24.6 MB/s eta 0:00:01
   -------------------------- ------------- 7.5/11.5 MB 24.0 MB/s eta 0:00:01
   ---------------------------------- ----- 10.0/11.5 MB 27.9 MB/s eta 0:00:01
   ---------------------------------------  11.5/11.5 MB 36.4 MB/s eta 0:00:01
   ---------------------------------------  11.5/11.5 MB 36.4 MB/s eta 0:00:01
   -----------

In [12]:
import pandas as pd

In [15]:
covid_df = pd.read_csv('italy-covid-daywise.csv')
covid_df

date  new_cases  new_deaths  new_tests
0    2019-12-31        0.0         0.0        NaN
1    2020-01-01        0.0         0.0        NaN
2    2020-01-02        0.0         0.0        NaN
3    2020-01-03        0.0         0.0        NaN
4    2020-01-04        0.0         0.0        NaN
..          ...        ...         ...        ...
243  2020-08-30     1444.0         1.0    53541.0
244  2020-08-31     1365.0         4.0    42583.0
245  2020-09-01      996.0         6.0    54395.0
246  2020-09-02      975.0         8.0        NaN
247  2020-09-03     1326.0         6.0        NaN

[248 rows x 4 columns]

Data from a CSV file will be stored in a DataFrame object

In [14]:
type(covid_df)

pandas.core.frame.DataFrame

Here's what the dataframe shows us:
- four daywise counts for Covid-19 in Italy
- Metrics reports are new cases, new deaths, and new tests
- 248 days worth of data (Dec 12, 2019 to Sep 3, 2020)

We can view basic information about a dataframe using .info() and .columns

In [16]:
covid_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 248 entries, 0 to 247
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   date        248 non-null    object 
 1   new_cases   248 non-null    float64
 2   new_deaths  248 non-null    float64
 3   new_tests   135 non-null    float64
dtypes: float64(3), object(1)
memory usage: 7.9+ KB


In [18]:
covid_df.columns

Index(['date', 'new_cases', 'new_deaths', 'new_tests'], dtype='object')

Each column contains a specific data type. For numeric columns, you can view statistical information (mean, standard deviation, min/max values, and number of non-empty values) using .describe()

In [17]:
covid_df.describe()

new_cases  new_deaths     new_tests
count   248.000000  248.000000    135.000000
mean   1094.818548  143.133065  31699.674074
std    1554.508002  227.105538  11622.209757
min    -148.000000  -31.000000   7841.000000
25%     123.000000    3.000000  25259.000000
50%     342.000000   17.000000  29545.000000
75%    1371.750000  175.250000  37711.000000
max    6557.000000  971.000000  95273.000000